File: android/app/src/main/java/com/mysteriumvpn/TestModule.java
package com.mysteriumvpn;

import android.util.Log;

import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

import java.util.Map;
import java.util.HashMap;

public class TestModule extends ReactContextBaseJavaModule {

    public TestModule(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @Override
    public String getName() {
        return "TestModule";
    }

    @Override
    public Map<String, Object> getConstants() {
        final Map<String, Object> constants = new HashMap<>();
        constants.put("foo", "bar");
        return constants;
    }

    @ReactMethod
    public void startService() {
        Log.d("TestModule", "Starting some sort of service");
    }
}

File: ios/New Group/TestModule.m
//
//  TestModule.m
//  MysteriumVPN
//
//  Created by Arnas Dundulis on 27/08/2018.
//  Copyright © 2018 Mysterium Network. All rights reserved.
//

#import "TestModule.h"
#import <React/RCTLog.h>

@implementation TestModule

RCT_EXPORT_MODULE();

- (NSDictionary *)constantsToExport
{
    return @{ @"foo": @"bar" };
}

RCT_EXPORT_METHOD(startService)
{
    RCTLogInfo(@"Starting some sort of service");
}

@end
